src/timely-util/src/containers.rs

```diff
@@ -126,12 +126,13 @@ mod container {
                 Column::Bytes(b) => {
                     assert_eq!(b.len() % 8, 0);
                     let mut alloc: Region<u64> = super::alloc_aligned_zeroed(b.len() / 8);
-                    bytemuck::cast_slice_mut(&mut alloc[..]).copy_from_slice(&b[..]);
+                    let alloc_bytes = bytemuck::cast_slice_mut(&mut alloc);
+                    alloc_bytes[..b.len()].copy_from_slice(b);
                     Self::Align(alloc)
                 }
                 Column::Align(a) => {
                     let mut alloc = super::alloc_aligned_zeroed(a.len());
-                    alloc.extend_from_slice(&a[..]);
+                    alloc[..a.len()].copy_from_slice(a);
```

Comment:
It's a bug, thanks for catching!

We might not want to call `alloc_aligned_zeroed` here because it sets the memory twice. Instead, we'd want to allocate an uninitialized region and use `extend_from_slice` as before. This needs a new API because `alloc_aligned_zeroed` checks whether to use lgalloc or not, and we don't have an equivalent non-initializing variant.

Reply:
Makes sense! I'd like to treat that out of scope for this PR though.